src/components/cdTab.js

```diff
@@ -1,34 +1,183 @@
+} from '@mui/material'
 
 import ObjectTextViewer from './util/ObjectTextViewer'
+import { useFetchComponentDescriptor } from '../api/useFetch'
+import CenteredSpinner from './util/CenteredSpinner'
+
+
+const FetchRawComponent = ({
+  setRaw,
+  setLoading,
+  setIsError,
+  componentName,
+  componentVersion,
+  ocmRepo,
+  versionFilter,
+}) => {
+  // eslint-disable-next-line no-unused-vars
+  const [componentDescriptor, isLoading, isError, _] = useFetchComponentDescriptor({
+    componentName: componentName,
+    ocmRepoUrl: ocmRepo,
+    version: componentVersion,
+    versionFilter: versionFilter,
+    raw: true,
+  })
+
+  React.useEffect(() => {
+    setRaw(componentDescriptor)
+    setLoading(isLoading)
+    setIsError(isError)
+  }, [
+    componentDescriptor,
+    isLoading,
+    isError,
+    setRaw,
+    setLoading,
+    setIsError,
+  ])
+}
+FetchRawComponent.displayName = 'FetchRawComponent'
+FetchRawComponent.propTypes = {
+  setRaw: PropTypes.func.isRequired,
+  setLoading: PropTypes.func.isRequired,
+  setIsError: PropTypes.func.isRequired,
+  componentName: PropTypes.string.isRequired,
+  componentVersion: PropTypes.string.isRequired,
+  ocmRepo: PropTypes.string,
+  versionFilter: PropTypes.string,
+}
+
+
+const Options = ({
+  isLoading,
+  raw,
+  setRaw,
+}) => {
+  return <Box>
+    <FormGroup>
+      <FormControlLabel
+        control={
+          <Tooltip
+            title='Raw document as retrieved from OCM repository, without (de)serialisation'
+          >
+            <Switch
+              checked={raw}
+              // eslint-disable-next-line no-unused-vars
+              onChange={(event) => setRaw((prev) => !prev)}
+              disabled={isLoading}
+            />
+          </Tooltip>
+        }
+        label='show raw'
+      />
+    </FormGroup>
+  </Box>
+}
+Options.displayName = 'Options'
+Options.propTypes = {
+  isLoading: PropTypes.bool,
+  raw: PropTypes.bool.isRequired,
+  setRaw: PropTypes.func.isRequired,
+}
+
+
+const Viewer = ({
+  componentDescriptor,
+  isLoading,
+  isError
+}) => {
+  if (isError) return <Alert severity='critical'>
+    Something went wrong
+  </Alert>
+
+  if (isLoading || !componentDescriptor) return <Box
+    display='flex'
+    justifyContent='center'
+  >
+    <CenteredSpinner/>
+  </Box>
+
+  return <ObjectTextViewer
+    obj={componentDescriptor}
+  />
+}
+Viewer.displayName = 'Viewer'
+Viewer.propTypes = {
+  isLoading: PropTypes.bool,
+  componentDescriptor: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  isError: PropTypes.bool,
+}
 
 
 export const CdTab = ({
   componentDescriptor,
   isLoading,
+  ocmRepo,
+  versionFilter,
 }) => {
-  if (!isLoading) return <ObjectTextViewer obj={componentDescriptor}/>
+  const [rawMode, setRawMode] = React.useState(false)
+
+  const [raw, setRaw] = React.useState()
+  const [rawLoading, setRawLoading] = React.useState()
+  const [rawIsError, setRawIsError] = React.useState()
+
+  const _isLoading = () => {
```

Comment:
I think the ternary operator would be less cumbersome here (and in the following two functions) while providing the same level of readability.